### PR TITLE
feat(CO-2747): Implement option to add external calendar via ics URL with sync

### PR DESCRIPTION
**Pull request by keshavbhatt**
- add context menu item to add external ICS calendar from URL
- add Add external ICS calendar from URL modal to allow user provide name and color for calendar to be imoroted
- add various validation checks in the add dialog
- show a icon on external calendar accordion item to distinguish them from other calendars 
- add edit external ICS calendar dialog to change the name and color of the calendar
- add a context menu action to trigger sync of calendar
- add a label on the sync action that shows last sync datetime
- snackbar messages for all operations - progress, success, failures.
- update the appoint view to see the appointments in external ICS calendar from owner's perspective
- show a icon on external calendar appointments items to distinguish them from other calendars appointments
- show a icon on external calendar appointment items in detail view
- show a icon on external calendar appointment items in quick summary view 

---

- expose lsd (last sync datetime) in mailbox GetFolder SOAP API
- expose new folder option called lsd (last sync datetime) from ui-commons
- make GetFolder SOAP API report correct flags for sync folders
- make GetFolder SOAP API report correct permission for sync folders

requires: https://github.com/zextras/carbonio-mailbox/pull/954 and https://github.com/zextras/carbonio-ui-commons/pull/254

_NOTE FOR REVIEWERS:_ 

Those +2k lines may look scary, they are probably due to package-lock.json update since we updated ui-commons library.

### CODE REVIEW

File: src/actions/calendar-actions-fn.tsx [hunk 1]
```diff
@@ -245,15 +296,49 @@ export const removeFromList =
...
+					key: `external-calendar-sync`,
+					replace: true,
+					severity: 'success',
+					hideButton: true,
+					label: t('message.snackbar.external_calendar_synced', 'Calendar synced successfully'),
+					autoHideTimeout: 3000
+				});
 			}
 		});
 	};
```
[the remark this answers is not in the document]
Comment (keshavbhatt) [reply src/actions/calendar-actions-fn.tsx:344]: This is outdated.

File: src/view/event-panel-view/organizer-part.tsx [hunk 2]
```diff
@@ -39,10 +40,22 @@ export const OrganizerPart = ({
...
 	const calendar = useFolder(invite.ciFolder);
 	const createSnackbar = useSnackbar();
-	const iAmAttendee = useMemo(
-		() => (!invite.isOrganizer && !(calendar as LinkFolder)?.owner) ?? false,
-		[calendar, invite.isOrganizer]
+	const isExternalCalendar = useMemo(() => isExternalSyncFolder(calendar ?? {}), [calendar]);
+	const isLoggedInUserAttendee = useMemo(
+		() =>
+			invite?.attendees?.some(
+				(attendee) => attendee?.a === account.name || attendee?.a === account.displayName
```
[the remark this answers is not in the document]
Comment (keshavbhatt) [reply src/view/event-panel-view/organizer-part.tsx:47]: Wont fix it, existing behavior.

File: src/actions/modals/edit-external-calendar-modal.tsx [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+	const folders = useFoldersMap();
+	const createSnackbar = useSnackbar();
+
+	const [calendarName, setCalendarName] = useState(folder?.name ?? '');
+	const [isSubmitting, setIsSubmitting] = useState(false);
+
+	// Initialize from rgb when color index is absent (typical for ICS-from-URL calendars).
+	const originalColorIndex = useMemo(() => resolveColorIndex(folder?.color, folder?.rgb), [folder]);
+	const [selectedColor, setSelectedColor] = useState(originalColorIndex);
+	const genericErrorLabel = t('label.error_try_again', 'Something went wrong, please try again');
```
[the remark this answers is not in the document]
Comment (keshavbhatt) [reply src/actions/modals/edit-external-calendar-modal.tsx:65]: We will never hit this corner case. user wont be able to open edit model if there is no such folder in store.

File: src/actions/appointment-actions-items.test.ts [hunk 4]
```diff
@@ -46,7 +46,42 @@ describe('edit event item', () => {
...
+
 	describe('is disabled when', () => {
+		test('the event has no organizer but the calendar is read-only (external ICS)', () => {
+			const folder = {
+				id: FOLDERS.CALENDAR,
+				l: '1',
+				name: 'Calendar',
+				view: 'appointment',
+				absFolderPath: '/'
+			};
```
[the remark this answers is not in the document]
Comment (keshavbhatt) [reply src/actions/appointment-actions-items.test.ts]: fixed

**Review by nubsthead — APPROVED** [review]

**Review by FrancescoGottardi — APPROVED** [review]

**Review by CerDennis — APPROVED** [review]
